tests/integration/security/citadel/namespace_targeting_test.go

```diff
@@ -0,0 +1,93 @@
+package citadel
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+
+	v1 "k8s.io/api/core/v1"
+	mv1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	caOverrideLabel    = "ca.istio.io/override"
+	caEnvLabel         = "ca.istio.io/env"
+	testNs             = "targeting-test"
+	serviceAccountName = "sample-service-account"
+)
+
+// TestNamespaceTargeting verifies that Citadel respects the designated namespace targeting labels.
+// current implementation is closely tied with Kubernetes concepts, but this is difficult to avoid
+// until Citadel itself is less closely tied with Kubernetes concepts
+func TestNamespaceTargeting(t *testing.T) {
+	framework.NewTest(t).Run(func(ctx framework.TestContext) {
+
+		env := ctx.Environment().(*kube.Environment)
+
+		citadelNamespace := ist.Settings().SystemNamespace
+		type namespaceLabels map[string]string
+		testCases := map[string]struct {
+			labels   namespaceLabels
+			targeted bool
+		}{
+			"override label set to false, env label matches => should not generate secret": {
+				labels: namespaceLabels{
+					caOverrideLabel: "false",
+				},
+				targeted: false,
+			},
+			"no override label present, env label matches => should generate secret": {
+				labels: namespaceLabels{
+					caEnvLabel: citadelNamespace,
+				},
+				targeted: true,
+			},
+			"no override label present, no env label present, default is active => should generate secret": {
+				labels:   namespaceLabels{},
+				targeted: true,
+			},
+			"override label false, env label matches => should not generate secret": {
+				labels: namespaceLabels{
+					caEnvLabel:      citadelNamespace,
+					caOverrideLabel: "false",
+				},
+				targeted: false,
+			},
+		}
+
+		for name, tc := range testCases {
+			ctx.NewSubTest(name).
+				Run(func(ctx framework.TestContext) {
+					doTest(t, env, ctx, tc.labels, tc.targeted)
+				})
+		}
+	})
+}
+
+func doTest(
```

Comment:
Consider the test pkg is `citadel`, and all the private methods are visible within the package. I would thus suggest to use more descriptive names.

```
func runNamespaceTargetCase(t *testing.T) {
  t.Helper();
}
```